llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -979,8 +987,10 @@ Value *getRuntimeVF(IRBuilderBase &B, Type *Ty, ElementCount VF) {
 }
 
 const SCEV *createTripCountSCEV(Type *IdxTy, PredicatedScalarEvolution &PSE,
-                                Loop *OrigLoop) {
-  const SCEV *BackedgeTakenCount = PSE.getBackedgeTakenCount();
+                                Loop *OrigLoop, bool IsSpeculative) {
+  const SCEV *BackedgeTakenCount = IsSpeculative
+                                       ? PSE.getSymbolicMaxBackedgeTakenCount()
```

Comment:
If there's a exact BTC then symbolic Max BTC == exact BTC, so the distinction here shouldn't be needed?